core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -457,7 +458,7 @@ trait WhiskWebActionsApi extends Directives with ValidateRequestSize with PostAc
    * This method is factored out to allow mock testing.
    */
   protected def getIdentity(namespace: EntityName)(implicit transid: TransactionId): Future[Identity] = {
-    Identity.get(authStore, namespace)
+    SpiLoader.get[AuthenticationDirectiveProvider].namespaceIdentity(namespace)(transid, authStore)
```

Comment:
You'll want to cache the call to `SpiLoader.get[AuthenticationDirectiveProvider]`. That's not something which should be executed on a per request basis.